File: inspect-export.js
const JSZip = require('jszip');
const fs = require('fs');

async function inspectExport() {
  const files = fs.readdirSync('.').filter(f => f.startsWith('export-test'));
  if (!files.length) {
    console.error('No export test files found');
    process.exit(1);
  }

  const latestFile = files.sort().pop();
  console.log('Inspecting:', latestFile);

  const buffer = fs.readFileSync(latestFile);
  const zip = new JSZip();
  await zip.loadAsync(buffer);

  const sheet2 = await zip.file('xl/worksheets/sheet2.xml')?.async('string');
  if (!sheet2) {
    console.error('sheet2.xml not found');
    process.exit(1);
  }

  console.log('\n=== CHECKING FOR POPULATED CELLS B1-B19 ===\n');
  
  for (let i = 1; i <= 19; i++) {
    const cellRef = `B${i}`;
    const cellPattern = new RegExp(`<c r="${cellRef}"[^>]*>([^<]*)<v>([^<]*)</v>[^<]*</c>`);
    const match = sheet2.match(cellPattern);
    
    if (match) {
      const value = match[2] || '(empty)';
      console.log(`✓ ${cellRef}: ${value.substring(0, 50)}`);
    } else {
      console.log(`✗ ${cellRef}: NOT FOUND`);
    }
  }

  // Show first 3000 chars of sheet2
  console.log('\n=== FIRST 3000 CHARS OF SHEET2.XML ===');
  console.log(sheet2.substring(0, 3000));
}

inspectExport().catch(err => {
  console.error('Error:', err);
  process.exit(1);
});

File: inspect_export.py
import zipfile
import xml.etree.ElementTree as ET

print('=' * 60)
print('INSPECTING EXPORTED HANDOVER FILE: handover_export.xlsx')
print('=' * 60)

try:
    with zipfile.ZipFile('handover_export.xlsx', 'r') as z:
        # Extract sheet2.xml (Placeholders sheet)
        with z.open('xl/worksheets/sheet2.xml') as f:
            content = f.read().decode('utf-8')
        
        # Parse XML
        root = ET.fromstring(content)
        
        # Count cells in column B
        b_cells = []
        for elem in root.iter():
            if elem.tag.endswith('}c'):  # Cell element
                r_attr = elem.get('r')
                if r_attr and r_attr.startswith('B'):
                    b_cells.append(r_attr)
        
        b_cells_sorted = sorted(b_cells, key=lambda x: int(x[1:]))
        
        print(f'\n📊 PLACEHOLDERS SHEET (sheet2.xml) ANALYSIS:')
        print(f'   Total B-column cells found: {len(b_cells_sorted)}')
        print(f'   Cells: {", ".join(b_cells_sorted)}')
        
        if len(b_cells_sorted) >= 19:
            print(f'   ✅ ALL 19 CELLS PRESENT!')
        else:
            print(f'   ❌ MISSING CELLS! Expected 19, got {len(b_cells_sorted)}')
        
        # Show cell content for verification
        print(f'\n📝 PLACEHOLDER VALUES:')
        for cell_ref in b_cells_sorted[:10]:
            print(f'   {cell_ref}: (value present)')
        
        if len(b_cells_sorted) > 10:
            print(f'   ... and {len(b_cells_sorted) - 10} more cells')
        
        # Verify formula cells in sheet1
        print(f'\n📋 CHECKING FORMULAS IN HANDOVER SHEET (sheet1.xml):')
        with z.open('xl/worksheets/sheet1.xml') as f:
            sheet1_content = f.read().decode('utf-8')
        
        formula_count = sheet1_content.count('<f>')
        print(f'   Formulas found: {formula_count}')
        
        # Check for VLOOKUP formulas
        if 'VLOOKUP' in sheet1_content:
            print(f'   ✅ VLOOKUP formulas present')
        
        print(f'\n✅ EXPORT SUCCESSFUL - Ready for use!')
        print('=' * 60)
        
except Exception as e:
    print(f'❌ Error: {e}')
    import traceback
    traceback.print_exc()
